### DISCUSSION TITLE
Change padding & gap dynamically?

### DISCUSSION THREAD

**Discussion by nepthar** [asmvik/yabai#802] [Ideas]
Hello!

I would like to be able to change padding and gap dynamically - If there is only one single window present, I'd like additional padding around it. But when I have multiple windows, I want all of the screen real estate possible.

Is it possible to do this today? If not, I'd be happy to try to implement this myself given a few pointers on where to dive in.

Thanks!

**Comment by rockyzhang24** [comment]
I wrote some demo pieces of code in this issue (https://github.com/koekeishiya/yabai/issues/784). Almost the same requirement with yours. I believe you will get the hints and inspiration.

**Reply by nepthar** [reply]
I tried this, but ended up backing out the change because gaps are global and not per-monitor. If I find the time, I'll try to submit a PR with the changes that would allow it.

**Comment by SxC97** [comment]
I have this exact feature enabled on my personal yabai and skhd config that you can find [here](https://github.com/SxC97/Yabai-SKHD-Config).  The code that you want to look at is in `.scripts/.gaps` and called in `.yabairc` via custom rules. Let me know if you have any questions getting it working!

**Reply by nepthar** [reply]
Thanks for taking a look. Does this work with multi monitor setups?

**Reply by SxC97** [reply]
Yes, the script sets the gaps to 0 per space instead of toggling gaps globally. Since displays have unique spaces, multi-monitors are supported.

Note: If you want to change the gaps variable, it has to be done in the `.gaps` script, not in your `.yabairc`.
